code/modules/mining/lavaland/ash_flora.dm

```diff
@@ -146,107 +147,108 @@
 	w_class = 1
 	resistance_flags = 0
 	burntime = 30
 	var/prepared = FALSE
-	var/prepared_type = /obj/item/ash_flora/shavings
+	var/prepared_type = /obj/item/weapon/reagent_containers/food/snacks/ash_flora/shavings
 	var/standard_prep = TRUE
 
-/obj/item/ash_flora/New()
+/obj/item/weapon/reagent_containers/food/snacks/ash_flora/New()
 	..()
 	pixel_x = rand(-4, 4)
 	pixel_y = rand(-4, 4)
 
-/obj/item/ash_flora/attackby(obj/item/weapon/W, mob/user, params)
-	if(!standard_prep || !W.sharpness || !prepare(user))
-		return ..()
 
-/obj/item/ash_flora/proc/prepare(mob/user)
+/* //
+/obj/item/weapon/reagent_containers/food/snacks/ash_flora/proc/prepared(mob/user)
 	if(!prepared)
 		prepared = TRUE
 		user << "<span class='notice'>You prepare [src].</span>"
-		var/obj/item/ash_flora/A = new prepared_type(get_turf(src))
+		var/obj/item/weapon/reagent_containers/food/snacks/ash_flora/A = new prepared_type(get_turf(src))
 		if(user.is_holding(src))
 			user.unEquip(src, TRUE)
 			user.put_in_hands(A)
 		qdel(src)
 		. = 1
 
-/obj/item/ash_flora/shavings //for actual crafting
+
+/obj/item/weapon/reagent_containers/food/snacks/ash_flora/attackby(obj/item/weapon/W, mob/user, params)
+	if(!standard_prep || !W.sharpness || !prepared(user))
+		return ..()
+*/
+
+/obj/item/weapon/reagent_containers/food/snacks/ash_flora/shavings //for actual crafting
 	prepared = TRUE
 
-/obj/item/ash_flora/mushroom_leaf
+/obj/item/weapon/reagent_containers/food/snacks/ash_flora/mushroom_leaf
 	name = "mushroom leaf"
-	desc = "A leaflike structure from a mushroom. You should probably wash the ash off."
+	desc = "A leaf, from a mushroom."
+	list_reagents = list("nutriment" = 3, "vitfro" = 2, "nicotine" = 2)
 	icon_state = "mushroom_leaf"
-	standard_prep = FALSE
-	prepared_type = /obj/item/ash_flora/mushroom_leaf/prepared
-
-/obj/item/ash_flora/mushroom_leaf/afterattack(atom/target, mob/user, proximity_flag)
+	//standard_prep = FALSE
+	//prepared_type = obj/item/weapon/reagent_containers/food/snacks/ash_flora/mushroom_leaf/prepared
+/* //
+/obj/item/weapon/reagent_containers/food/snacks/ash_flora/mushroom_leaf/afterattack(atom/target, mob/user, proximity_flag)
 	if(proximity_flag && (istype(target, /turf/open/floor/plating/ashplanet/wateryrock) || istype(target, /obj/structure/sink)))
-		prepare(user)
-
-/obj/item/ash_flora/mushroom_leaf/prepared
+		prepared(user)
+*/
+/obj/item/weapon/reagent_containers/food/snacks/ash_flora/mushroom_leaf/prepared
 	name = "washed mushroom leaf"
 	desc = "A greenish leaflike structure from a mushroom. Kind of like lettuce?"
 	icon_state = "mushroom_leaf_p"
 	prepared = TRUE
 
-/obj/item/ash_flora/mushroom_cap
+/obj/item/weapon/reagent_containers/food/snacks/ash_flora/mushroom_cap
 	name = "mushroom cap"
-	desc = "A mushroom cap. Possibly delicious, at least if you cut it up a bit first."
+	desc = "The cap of a large mushroom."
+	list_reagents = list("mindbreaker" = 2, "entpoly" = 4, "mushroomhallucinogen" = 2)
```

Comment:
"Has use for traitors provided the traitor has the ability to actually get the toxin" isn't particularly good either, as AS IT HAPPENS most people who go on lavaland are either not traitors or can't exploit the toxin.